classes/timeslot_manager.php

```diff
@@ -0,0 +1,282 @@
+            if ($data->duration < 1) {
+                throw new \coding_exception("Property duration cannot be below one");
+            }
+        }
+
+        if ($newinstance) {
+            // Insert.
+            return $DB->insert_record($tablename, $data, $returnid);
+        } else {
+            return $DB->update_record($tablename, $data);
+        }
+    }
+
+    /**
+     * Gets observation point data
+     * @param int $observationid ID of observation instance
+     * @param int $slotid ID of the observation point
+     * @param string $tablename database table name
+     * @return object existing point data
+     */
+    public static function get_existing_slot_data(int $observationid, int $slotid,
+            string $tablename = 'observation_timeslots'): object {
+        global $DB;
+        return $DB->get_record($tablename, ['id' => $slotid, 'obs_id' => $observationid], '*', MUST_EXIST);
+    }
+
+    /**
+     * Get all observation points in an observation instance
+     * @param int $observationid ID of the observation instance
+     * @param string $sortby column to sort by
+     * @param string $tablename database tablename
+     * @return array array of database objects obtained from database
+     */
+    public static function get_time_slots(int $observationid, string $sortby='list_order',
+            string $tablename='observation_timeslots'): array {
+        global $DB;
+        return $DB->get_records($tablename, ['obs_id' => $observationid], $sortby);
+    }
+
+    /**
+     * Deletes observation point
+     * @param int $observationid ID of the observation instance
+     * @param int $obpointid ID of the observation point to delete
+     * @param string $tablename database table name
+     */
+    public static function delete_time_slot(int $observationid, int $obpointid, string $tablename='observation_timeslots') {
+        global $DB;
+        // To ensure ordering stays intact, should move all those points with a higher order than this one down by 1.
+        $currentpoint = self::get_existing_point_data($observationid, $obpointid, $tablename);
+
+        // Get those with a higher list ordering than this one.
+        $pointsabove = $DB->get_records_select(
+            $tablename,
+            "obs_id = :obsid AND list_order > :listorder",
+            [
+                'obsid' => $observationid,
+                'listorder' => $currentpoint->list_order
+            ]
+        );
+
+        $transaction = $DB->start_delegated_transaction();
+
+        $DB->delete_records($tablename, ['id' => $obpointid, 'obs_id' => $observationid]);
+
+        // Shuffle those above down.
+        foreach ($pointsabove as $pointabove) {
+            $DB->update_record(
+                $tablename,
+                [
+                    'id' => $pointabove->id,
+                    'list_order' => $pointabove->list_order - 1
+                ]
+            );
+        }
+
```

Comment:
Seems to be more artefacts from copying observation points - observation timeslots don't have a list_order do they ?
Also the function is delete_time_slot, but it appears to actually update them rather than delete them.

Reply:
correct, have removed everything I think is unneeded, this photo shows the lines that have remained without anything to do with the list numbers because they are only for observation points
![image](https://user-images.githubusercontent.com/56417228/128444464-6f648d8a-1cb5-4471-8187-8151e3db0fed.png)